workspaces/ui-v2/src/pages/changelog/ChangelogEndpointRootPage.tsx

```diff
@@ -49,20 +50,31 @@ const ChangelogRootComponent: FC<
     batchId: string;
   }>
 > = ({ match }) => {
-  const { endpoints, loading } = useEndpoints();
+  const endpointsState = useAppSelector((state) => state.endpoints.data);
+
+  const dispatch = useAppDispatch();
+  const spectacle = useSpectacleContext();
+  useEffect(() => {
+    // the typing is correct, we just arent setting it correctly above
+    // TODO remove ts-ignore
+    // @ts-ignore
+    dispatch(endpointActions.fetchEndpoints({ spectacle }));
+  }, [dispatch, spectacle]);
```

Comment:
Also note, this _should_ cause this action to be refetched if the spectacle instance changes (which is what we want). There are two options we have to handle the old data
- send a reload actions that all redux stores reset to loading
- just let all fetch.fulfilled actions overwrite the old store (we need to make sure we build the store with this requirement in mind)